src/dmd/lexer.d

```diff
@@ -202,6 +203,104 @@ unittest
     }
 }
 
+
+/**************
+ * Pool allocator for Token objects, used by the Lexer.
+ * Tokens are never freed so Token pointers will never change
+ */
+private struct TokenPool
+{
+    private Array!Token tokens;
+
+    public ref Token getByIndex(size_t index) nothrow pure
+    {
+        if (index >= tokens.length)
+        {
+            tokens.setDim(index + 1);
+        }
+        return tokens[index];
+    }
+}
+
+
+/***************
+ * Represents a range of tokens lazily scanned from a lexer
+ * should be created with the Lexer makeRangeFromHere method
+ */
+struct TokenRange
+{
+private:
+    Lexer lex; //where are these tokens coming from?
```

Comment:
`lex` sounds more like a `verb`. I think `lexer` would be a better name.

Reply:
I can't rename both the member var and the method to lexer.  Is there a dmd convention for naming "shadowed" members?

Reply:
I append `_` to the name. Some will prepend it instead.

Reply:
Walter told me "Please don't use _ in identifier names unless it is a postfix for a keyword name."

Reply:
I know. But now there's a reason. Not sure if he prefers `_` to be prepended.